### PR TITLE
Hlasm listing grammer

### PR DESCRIPTION
**Pull request by LukasJaks**
added support for hlasm listing

### CODE REVIEW

**Comment by slavek-kucera** [comment]
any ideas on how to add a few tests?

**Comment by slavek-kucera** [comment]
Also, I see a few issues
1) there seems to be "off by one" problem when coloring listings from Endevor (or other formats with ASA control characters),
2) it looks like the full line comments (line starts with `*` or `.*`) are not recognized.
3) The page title seems to be recognized as a regular line (but I am not sure at the moment if this can be solved).

**Comment by LukasJaks** [comment]
off by one for Endevor listing with ASA control characters is fixed now.
Didn't manage to make multiline comments to work. Have still the same issue that it's only passing one line at a time.
I wasn't able to replicate this issue.

**Comment by slavek-kucera** [comment]
the `lines_to_ignore` need their endevor variant as well due to the extra character (or perhaps they can be formulated to work in both cases).

**Comment by LukasJaks** [comment]
Sorry, forgot. it's fixed now. Right now there's a variant, will look into if I can make them work for both.

**Comment by slavek-kucera** [comment]
it is fine to have two variants... it actually might more flexible if we discover that we need to ignore additional lines in the Endevor case.

**Comment by slavek-kucera** [comment]
But I am still seeing the issue with comments. Lines starting with '*' are still highlighted as regular line with label, instruction, operands etc.

File: clients/vscode-hlasmplugin/syntaxes/hlasmListing.tmLanguage.json [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+    {
+      "begin": "(.)  Loc    Object Code      Addr1    Addr2    Stmt  Source Statement",
+      "end": "                                                 Relocation Dictionary",
+      "patterns": [ {"include": "#lines_to_ignore_endevor"}, {"include" : "#listing_syntax_endevor_long"}]
+    },
+    {
+      "begin": "(.)  Loc  Object Code    Addr1 Addr2  Stmt   Source Statement",
+      "end": "                                                 Relocation Dictionary",
+      "patterns": [ {"include": "#lines_to_ignore_endevor"}, {"include" : "#listing_syntax_endevor"}]
+    }
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/syntaxes/hlasmListing.tmLanguage.json:25]:
```suggestion
    {
      "begin": "(.)  Loc    Object Code      Addr1    Addr2    Stmt  Source Statement",
      "end": "(.)                                                 Relocation Dictionary",
      "patterns": [ {"include": "#lines_to_ignore_endevor"}, {"include" : "#listing_syntax_endevor_long"}]
    },
    {
      "begin": "(.)  Loc  Object Code    Addr1 Addr2  Stmt   Source Statement",
      "end": "(.)                                                 Relocation Dictionary",
      "patterns": [ {"include": "#lines_to_ignore_endevor"}, {"include" : "#listing_syntax_endevor"}]
    }
```

File: clients/vscode-hlasmplugin/syntaxes/hlasmListing.tmLanguage.json [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+    {
+      "begin": "(.)  Loc    Object Code      Addr1    Addr2    Stmt  Source Statement",
+      "end": "(.)                                                 Relocation Dictionary",
+      "patterns": [ {"include": "#lines_to_ignore_endevor"}, {"include" : "#listing_syntax_endevor_long"}]
+    },
+    {
+      "begin": "(.)  Loc  Object Code    Addr1 Addr2  Stmt   Source Statement",
+      "end": "(.)                                                 Relocation Dictionary",
+      "patterns": [ {"include": "#lines_to_ignore_endevor"}, {"include" : "#listing_syntax_endevor"}]
+    }
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/syntaxes/hlasmListing.tmLanguage.json:25]: It looks like Endevor places the member name at the beginning of the line, so let's relax the end condition a bit
```suggestion
    {
      "begin": "(.)  Loc    Object Code      Addr1    Addr2    Stmt  Source Statement",
      "end": "(1.{8})                                         Relocation Dictionary",
      "patterns": [ {"include": "#lines_to_ignore_endevor"}, {"include" : "#listing_syntax_endevor_long"}]
    },
    {
      "begin": "(.)  Loc  Object Code    Addr1 Addr2  Stmt   Source Statement",
      "end": "(1.{8})                                         Relocation Dictionary",
      "patterns": [ {"include": "#lines_to_ignore_endevor"}, {"include" : "#listing_syntax_endevor"}]
    }
```

**Review by slavek-kucera — APPROVED** [review]

**Pull request merged**